### PR TITLE
Fix 32-bit Linux build on 64-bit host system.

### PR DESCRIPTION
**Pull request by weitjong**
Using pkg-config based detection is not reliable when cross-compiling. The proposed solution assumes when the development package is installed then its corresponding header file is installed and it can be searched using a list of known search paths that work for both 32/64-bit arch.

Additionally, I have also fixed the FIXME to bake the `SDL_USE_IME` define in the `SDL_config.h.cmake` file.

The patch is CI tested successfully here on Linux runner: https://github.com/alpha-scorpii-b/SDL/actions/runs/1130945633

### CODE REVIEW

**Comment by weitjong** [comment]
I have to force push to remove a redundant variable check in the earlier version.

**Comment by smcv** [comment]
Please don't do this. The official way to find at least GLib and libdbus is with pkg-config; the precise paths that are mentioned in the .pc file are considered to be an implementation detail and may change without notice.

There are two ways to make pkg-config multilib- or cross-friendly:

* Have a separate pkg-config per architecture, conventionally named something like `x86_64-pc-linux-gnu-pkg-config` and `i686-pc-linux-gnu-pkg-config`. Autotools will automatically try to use the architecture-specific `something-pkg-config` before falling back to plain `pkg-config`.

* Set `PKG_CONFIG_PATH` and/or `PKG_CONFIG_LIBDIR` to point to the `.pc` files installed by your 32-bit libraries, for example `PKG_CONFIG_LIBDIR=/usr/lib32/pkgconfig:/usr/share/pkgconfig`.

(Debian and Debian-derived OSs do both, in fact: `i686-pc-linux-gnu-pkg-config` on x86_64 Debian is a shell script that sets `PKG_CONFIG_LIBDIR` and then runs the normal `pkg-config`.)

File: CMakeLists.txt [hunk 1]
```diff
@@ -1282,28 +1282,43 @@ elseif(UNIX AND NOT APPLE AND NOT ANDROID AND NOT RISCOS AND NOT HAIKU)
...
-        # Fcitx need only dbus.
-        set(HAVE_FCITX TRUE)
-      endif()
+    # Cater for both multilib and multiarch, native and cross-compiling build
+    find_path(DBUS_H_INCLUDE_DIRS NAMES dbus/dbus.h PATH_SUFFIXES dbus-1.0)
+    if(ARCH_64)
+      set(DBUS_INC_SEARCH_PATH /usr/lib64/dbus-1.0/include /usr/lib/x86_64-linux-gnu/dbus-1.0/include)
+    else()
+      set(DBUS_INC_SEARCH_PATH /usr/lib/dbus-1.0/include /usr/lib/i386-linux-gnu/dbus-1.0/include)
+    endif()
```
Comment (smcv) [inline CMakeLists.txt:1291]: As well as making assumptions about libdbus implementation details, this also assumes everything is x86.

Comment (weitjong) [reply CMakeLists.txt:1291]: This is a valid point. The patch was originally made for my project with subset of target platforms in mind.

**Comment by weitjong** [comment]
Thanks for the very valuable information about the pkg-config. It seems I have to try harder to get it work with CMake when cross-compiling.

**Comment by weitjong** [comment]
Got it working by additionally installing `pkg-config-i686-linux-gnu` package on my 64-bit Linux container and by setting the `PKG_CONFIG` env-var to point to the newly installed wrapper script.

I will probably submit another PR later just for the below.
> Additionally, I have also fixed the FIXME to bake the SDL_USE_IME define in the SDL_config.h.cmake file.

**Pull request closed without merging**